File: go/stack/stack.go
package main

import "fmt"

type Stack []int

func (s *Stack) Push(v int) {
	*s = append(*s, v)
}

func (s *Stack) Pop() (int, bool) {
	if len(*s) == 0 {
		return 0, false
	} else {
		index := len(*s) - 1
		element := (*s)[index]
		*s = (*s)[:index]
		return element, true
	}
}

func main() {
	var stack Stack
	stack.Push(10)
	stack.Push(20)
	if val, ok := stack.Pop(); ok {
		fmt.Println(val) // 20
	}
	if val, ok := stack.Pop(); ok {
		fmt.Println(val) // 10
	}
	if _, ok := stack.Pop(); !ok {
		fmt.Println("Stack isEmpty!")
	}
}

File: java/src/stack/Stack.java
package stack ;

import java.util.ArrayList;

public class Stack {
    private ArrayList<Integer> stack = new ArrayList<>();

    public void push(int value) {
        stack.add(value);
    }

    public Integer pop() {
        if (stack.isEmpty()) {
            return null;
        } else {
            return stack.remove(stack.size() - 1);
        }
    }

    public static void main(String[] args) {
        Stack stack = new Stack();
        stack.push(10);
        stack.push(20);
        Integer val;
        if ((val = stack.pop()) != null) {
            System.out.println(val); // 20
        }
        if ((val = stack.pop()) != null) {
            System.out.println(val); // 10
        }
        if (stack.pop() == null) {
            System.out.println("Stack isEmpty!");
        }
    }
}

